### ISSUE TITLE
The container used by the client is different from the container used by the db manager in Function BaseTestCase

### ISSUE THREAD

**Issue by benglass** [symfony-cmf/Testing#84]
@dbu @dantleech @briancappello

We discussed in https://github.com/symfony-cmf/MenuBundle/pull/215 that there are 2 different containers which are created when you have a functional test that extends the BaseTestCase

1. The client creates a container which is used when processing requests
2. Any calls to $this->getContainer or $this->db($type) use a different container

This can cause issues with the 2 containers getting out of sync, for example if you delete a PHPCR node in a functional test and then try to find it in the DB to ensure its not there you can inadvertently end up using 2 different containers and trigger a PHPCR\InvalidItemStateException

this->db() proxies to this->getContainer, which creates a client and gets the container from it and throws it away.

https://github.com/symfony-cmf/Testing/blob/master/src/Functional/BaseTestCase.php#L58

Instead we could store an internal reference to the client as follows

```php
    public function getContainer()
    {
        if (!$this->client) {
            $this->client = $this->createClient($this->getKernelConfiguration());
        }
        $this->container = $this->client->getContainer();

        return $this->container;
    }
```

**Comment by WouterJ** [comment]
This is fixed in 1.3 by https://github.com/symfony-cmf/Testing/pull/109

**Comment by benglass** [comment]
@WouterJ awesome thanks!
